### PR TITLE
Add pagination settings

### LINKED ISSUE #294: Add support to the pagination setting customization at the index level
⚠️ This issue is generated, which means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Related to:
- parent issue: https://github.com/meilisearch/integration-guides/issues/205
- engine issue: https://github.com/meilisearch/specifications/pull/157, https://github.com/meilisearch/meilisearch/issues/2495

----

Add a `pagination` index settings resource to manage customization of the maximum number of reachable documents during the search at the index level.

- Expose `pagination` object on `/indexes/:index_uid/settings` endpoints.
- Add `GET/PATCH/DELETE` - `/indexes/:index_uid/settings/pagination` endpoints.

The `pagination` object definition is made of the following properties:
- `maxTotalHits` type (`int`)

More information about what each field does could be [checked in the spec.](https://github.com/meilisearch/specifications/pull/117)

Check the other customization methods for reference like: `synonyms`, `searchableAttributes`, `filterableAttributes`.

TODO:
- [ ] Add the new methods to customize `pagination`
  - [ ] GET `/indexes/:index_uid/settings/pagination`
  - [ ] PATCH `/indexes/:index_uid/settings/pagination`
  - [ ] DELETE `/indexes/:index_uid/settings/pagination`
- [ ] Add tests

### PR DESCRIPTION
**Pull request by alrz1999**
# Pull Request

## Related issue
Fixes #294 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by alallema** [comment]
Hello @alrz1999,
Thank you very much for contributing to Meilisearch :heart:.
However, I am not available on the weekend, but I will be back on Monday :blush:.

*PS: This message was sent automatically!*

File: tests/Meilisearch.Tests/SettingsTests.cs [hunk 1]
```diff
@@ -438,6 +442,49 @@ public async Task ResetTypoTolerance()
...
+                MaxTotalHits = 20
+            };
+
+            var returnedPagination = new Pagination
+            {
+                MaxTotalHits = 20
+            };
+
+            await AssertUpdateSuccess(_index.UpdatePaginationAsync, newPagination);
+            await AssertGetEquality(_index.GetPaginationAsync, returnedPagination);
```
Comment (alallema) [inline tests/Meilisearch.Tests/SettingsTests.cs]: Can't we use the same one twice `newPagination`?

File: tests/Meilisearch.Tests/SettingsTests.cs [hunk 2]
```diff
@@ -438,6 +442,49 @@ public async Task ResetTypoTolerance()
...
+                MaxTotalHits = 30
+            };
+
+            var returnedPagination = new Pagination
+            {
+                MaxTotalHits = 30
+            };
+
+            await AssertUpdateSuccess(_index.UpdatePaginationAsync, newPagination);
+            await AssertGetEquality(_index.GetPaginationAsync, returnedPagination);
```
Comment (alallema) [inline tests/Meilisearch.Tests/SettingsTests.cs]: Here too just used `newPagination` in the both tests?

**Review by alallema — COMMENTED** [review]
Thank @alrz1999 for this PR!
Well done 🔥  I just have one question about the tests
And so great to add the code-sample too ❤️

File: tests/Meilisearch.Tests/SettingsTests.cs (hunk 1 above)
Comment (alrz1999) [reply tests/Meilisearch.Tests/SettingsTests.cs]: Yes. Fixed.

File: tests/Meilisearch.Tests/SettingsTests.cs (hunk 2 above)
Comment (alrz1999) [reply tests/Meilisearch.Tests/SettingsTests.cs]: Fixed.

**Review by alallema — APPROVED** [review]
LGTM!
Thanks again for contributing with Meilisearch :heart:
Don't forget to complete [this form](https://meilisearch.typeform.com/to/idt5qugY) in case you didn't do it yet if you want to receive some swag!

**Comment by alallema** [comment]
bors merge

**Comment by alallema** [comment]
_This message is sent automatically_

Thanks again for contributing to Meilisearch :heart:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive some gift from Meilisearch too, please complete [this form](https://meilisearch.typeform.com/to/idt5qugY).

**Pull request merged**